### ISSUE TITLE
Boost cmake: why use Boost_NO_BOOST_CMAKE ON on macOs?

### ISSUE THREAD

**Issue by peter-d** [sparcians/map#395]
(triggered by an Olympia build issue https://github.com/riscv-software-src/riscv-perf-model/issues/35)

I've noticed there's a bit of a non-standard way of finding boost for Sparta fro macOs: 
https://github.com/sparcians/map/blob/edc3b2a2285f6c36b2a5c8311b147649959b6926/sparta/cmake/sparta-config.cmake#L50

Which makes it need ```-DBoost_DIR=/opt/homebrew/Cellar/boost@1.76/1.76.0_3/lib/cmake/Boost-1.76.0``` on the command line to find the boost paths if they are not in a system default location.

But most other repos (like stf_lib, for example), to find boost with cmake, you define ```-DBOOST_ROOT=/opt/homebrew/Cellar/boost@1.76/1.76.0_3```

Removing that line,  on my system at least, boost is still found correctly with the standard ```-DBOOST_ROOT=<path>```, but then I am not using conda...

Is there a specific reason to keep this like that, or can the boost setup be made simpler (and more uniform)?

**Comment by knute-sifive** [comment]
Getting boost working with MacOS/cmake a few years back was a nightmare.  Brew helps a bit, but there was something non-standard with how it was installed and how `cmake` interacted with it.  Perhaps this is now moot since we've moved _way_ past boost 1.65.  I think I took the attitude -- it ain't broke, so don't fix.

Now it's back to broke and not standard.  I'm ok with removing those constraints and trying again.
